### ISSUE TITLE
Unstable keyword should trigger when unstable configs are set via the command line

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#20680]
Missed this when Nikki did her initial implementation because writing a test for it is a right pain, with the combination of needing `--warn-unstable` to be thrown and interacting with setting the config at compilation time or execution time.

This should be pretty straight-forward to do, I'd expect it to look similar to #17848.  The only potentially tricky part is checking if the executable was built with `--warn-unstable` and keying off that for whether to warn when the config is set via an execution time flag.

**Comment by bradcray** [comment]
Do we have unstable configs today?

**Comment by lydia-duncan** [comment]
You can apply `@unstable` to configs, but I don't think we have actual configs that are marked unstable today

**Comment by lydia-duncan** [comment]
Andy noticed on #21150 that there are quite a few configs in the [Memory.Diagnostics module](https://chapel-lang.org/docs/latest/modules/standard/Memory/Diagnostics.html), which is not intended to be stabilized for 2.0.  I think that boosts the priority of this issue (though there may be additional work since configs should also print the warning if the module they are defined in is marked unstable, which should be the case for the Memory.Diagnostics module but may not be yet).

Looking at the help output, it also seems like it would be a good idea for an unstable config to mention that in the help output (even if they're only marked unstable because the module they live in is unstable).  Maybe something like:
```
[...]
CONFIG VARS:
============
Built-in config vars:
                                     [...]
                         memTrack: bool (unstable)
```
?

**Comment by lydia-duncan** [comment]
In our discussion on Tuesday about unstable configs:

Michael was concerned we can't add new configs in the future in a way that is non-breaking
- This is partially because doing so will impact a config in user code with the same name, but also because configs take precedence over generalized argument parsing.


Michael proposed that the way to provide unstable/deprecated configs in the `--help` output would be to utilize an additional flag (name TBD).  This flag would be mentioned at the bottom of the `--help` output.

Config variables should indicate their deprecation/unstable status in the --help output:
- Jada, Anna, Vass, Michael and I support this, Andy and Engin support it with concerns

Unstable/deprecated configs should not be included in the --help output in the first place:
- Jeremiah, Ben H, Ben M, David I, Jade, Engin, Brad and Ahmad support this, Vass and Michael support it with concerns

Don't print either deprecated or unstable by default; have a flag that prints them when opted in:
- Ahmad, Andy, Jade, Ben H, Jeremiah, Michael, Brad and I were in favor of this.  Vass was strongly opposed - he finds the idea of using an additional flag a lot of work.

Vass wanted to separate out the discussion of unstable and deprecated configs:

Unstable configs should be included in --help:
- David I, Vass, Anna and I were in favor of this

Deprecated configs should be included in --help:
- Engin, David I, Anna and Vass were strongly opposed, Andy and I support it with concerns, and Michael supports it.

Counterproposal to having a flag 1: have the unstable/deprecated symbol in a separate section
- I am in favor of this

Counterproposal 2: Have small indicator on symbol itself, e.g `memStats: bool (unstable)`:
- David L, Vass, Anna and I are in favor of this

Have a flag to opt into printing internal configs in --help
- David I, Anna, Vass, Andy, David L, Brad, and I are in favor of this.


Andy pointed out that the `--help` output is visible not just to Chapel programmers but to people that use the programs built by the Chapel compiler, and so might not know where to look to get more information about flags or find them.

Ahmad would like us to both have the separate flag and have a small indicator (like `(unstable)`) on the unstable symbols themselves.

----------

Summary:
- The consensus is that we will hide unstable and deprecated configs from the help output, enabling them via a separate flag, whose name is TBD
- We also will add a separate flag for including configs from the internal modules in the `--help` output

**Comment by e-kayrakli** [comment]
@lydia-duncan [pointed out](https://github.com/chapel-lang/chapel/issues/21150#issuecomment-1540710022) that we can hide some GPU configs by making them unstable. That's aligned with the general approach summarized above. Currently they are:

- `debugGpu`
- `enableGpuP2P`
- https://github.com/chapel-lang/chapel/pull/22172 is working towards adding something like `gpuNoCpuModeWarning`.

All of these are in `ChapelGpuSupport` module and should be marked as unstable. We don't mark everything unstable individually, but you get one unstablity warning for using the `gpu` locale model in general. Hiding them from `--help` output is a good motive for marking these unstable individually.

**Comment by lydia-duncan** [comment]
This was resolved by @e-kayrakli's #22849
